cli/src/main/java/com/devonfw/tools/ide/tool/GlobalToolCommandlet.java

```diff
@@ -24,13 +26,70 @@ public abstract class GlobalToolCommandlet extends ToolCommandlet {
    * @param context the {@link IdeContext}.
    * @param tool the {@link #getName() tool name}.
    * @param tags the {@link #getTags() tags} classifying the tool. Should be created via {@link Set#of(Object) Set.of}
-   * method.
+   *        method.
    */
   public GlobalToolCommandlet(IdeContext context, String tool, Set<Tag> tags) {
 
     super(context, tool, tags);
   }
 
+  /**
+   * Performs the installation of the {@link #getName() tool} via a package manager.
+   *
+   * @param silent - {@code true} if called recursively to suppress verbose logging, {@code false} otherwise.
+   * @param commands - A {@link Map} containing the commands used to perform the installation for each package manager.
+   * @return {@code true} if the tool was newly installed, {@code false} if the tool was already installed before and
+   *         nothing has changed.
+   */
+  protected boolean installWithPackageManger(Map<PackageManager, List<String>> commands, boolean silent) {
+
+    Path binaryPath = this.context.getPath().findBinary(Path.of(getBinaryName()));
+
+    if (binaryPath != null && Files.exists(binaryPath) && !this.context.isForceMode()) {
+      IdeLogLevel level = silent ? IdeLogLevel.DEBUG : IdeLogLevel.INFO;
+      this.context.level(level).log("{} is already installed at {}", this.tool, binaryPath);
+      return false;
+    }
+
+    Path bashPath = this.context.getPath().findBinary(Path.of("bash"));
+    if (bashPath == null || !Files.exists(bashPath)) {
+      context.warning("Bash was not found on this machine. Not Proceeding with installation of tool " + this.tool);
+      return false;
+    }
+
+    PackageManager foundPackageManager = null;
+    for (PackageManager pm : commands.keySet()) {
+      if (Files.exists(this.context.getPath().findBinary(Path.of(pm.toString().toLowerCase())))) {
+        foundPackageManager = pm;
+        break;
+      }
+    }
+
+    int finalExitCode = 0;
+    if (foundPackageManager == null) {
+      context.warning("No supported Package Manager found for installation");
+      return false;
+    } else {
+      List<String> commandList = commands.get(foundPackageManager);
+      if (commandList != null) {
+        for (String command : commandList) {
+          ProcessContext pc = this.context.newProcess().errorHandling(ProcessErrorHandling.WARNING).executable(bashPath)
+              .addArgs("-c", command);
```

Comment:
why do we need `bash -c ` to run the command?
OK, later down I found out that we are passing pipes (`echo ... | dd ...`) what only works this way so this is fully correct.
I still would like to simplify the list to a single command that can then be a "script" potentially calling multiple commands (`foo && bar`).